src/core/symbology/classification/qgsclassificationmethod.cpp

```diff
@@ -0,0 +1,300 @@
+
+  return method;
+}
+
+QDomElement QgsClassificationMethod::save( QDomDocument &doc, const QgsReadWriteContext &context ) const
+{
+  QDomElement methodElem = doc.createElement( QStringLiteral( "classificationMethod" ) );
+
+  methodElem.setAttribute( QStringLiteral( "id" ), id() );
+
+  // symmetric
+  QDomElement symmetricModeElem = doc.createElement( QStringLiteral( "symmetricMode" ) );
+  symmetricModeElem.setAttribute( QStringLiteral( "enabled" ), symmetricModeEnabled() ? 1 : 0 );
+  symmetricModeElem.setAttribute( QStringLiteral( "symmetrypoint" ), symmetryPoint() );
+  symmetricModeElem.setAttribute( QStringLiteral( "astride" ), mSymmetryAstride ? 1 : 0 );
+  methodElem.appendChild( symmetricModeElem );
+
+  // label format
+  QDomElement labelFormatElem = doc.createElement( QStringLiteral( "labelFormat" ) );
+  labelFormatElem.setAttribute( QStringLiteral( "format" ), labelFormat() );
+  labelFormatElem.setAttribute( QStringLiteral( "labelprecision" ), labelPrecision() );
+  labelFormatElem.setAttribute( QStringLiteral( "trimtrailingzeroes" ), labelTrimTrailingZeroes() ? 1 : 0 );
+  methodElem.appendChild( labelFormatElem );
+
+  // extra information
+  QDomElement extraElem = doc.createElement( QStringLiteral( "extraInformation" ) );
+  saveExtra( extraElem, context );
+  methodElem.appendChild( extraElem );
+
+  return methodElem;
+}
+
+
+void QgsClassificationMethod::setSymmetricMode( bool enabled, double symmetryPoint, bool astride )
+{
+  mSymmetricEnabled = enabled;
+  mSymmetryPoint = symmetryPoint;
+  mSymmetryAstride = astride;
+}
+
+void QgsClassificationMethod::setLabelPrecision( int precision )
+{
+  // Limit the range of decimal places to a reasonable range
+  precision = qBound( MIN_PRECISION, precision, MAX_PRECISION );
+  mLabelPrecision = precision;
+  mLabelNumberScale = 1.0;
+  mLabelNumberSuffix.clear();
+  while ( precision < 0 )
+  {
+    precision++;
+    mLabelNumberScale /= 10.0;
+    mLabelNumberSuffix.append( '0' );
+  }
+}
+
+QString QgsClassificationMethod::formatNumber( double value ) const
+{
+  if ( mLabelPrecision > 0 )
+  {
+    QString valueStr = QLocale().toString( value, 'f', mLabelPrecision );
+    if ( mLabelTrimTrailingZeroes )
+      valueStr = valueStr.remove( RE_TRAILING_ZEROES );
+    if ( RE_NEGATIVE_ZERO.exactMatch( valueStr ) )
+      valueStr = valueStr.mid( 1 );
+    return valueStr;
+  }
+  else
+  {
+    QString valueStr = QLocale().toString( value * mLabelNumberScale, 'f', 0 );
+    if ( valueStr == QLatin1String( "-0" ) )
+      valueStr = '0';
+    if ( valueStr != QLatin1String( "0" ) )
+      valueStr = valueStr + mLabelNumberSuffix;
+    return valueStr;
+  }
+}
+
+QList<QgsClassificationRange> QgsClassificationMethod::classes( const QgsVectorLayer *vl, const QString &expression, int nclasses )
+{
+  if ( expression.isEmpty() )
+    return QList<QgsClassificationRange>();
+
+  if ( nclasses < 1 )
+    nclasses = 1;
+
+  QList<double> values;
+  double minimum;
```

Comment:
= std::numeric _limits...

Reply:
it is set a few lines after.
What would be the reasoning to set it?

Reply:
any uninitialised bool/int/double makes me nervous -- it just looks wrong...